pkg/client/mapiserver/snapshot.go

```diff
@@ -49,135 +46,53 @@ type SnapshotInfo struct {
 
 // CreateSnapshot creates a snapshot of volume by API request to m-apiserver
 func CreateSnapshot(volName string, snapName string, namespace string) error {
-	_, err := GetStatus()
-	if err != nil {
-		return err
-	}
-
-	var snap v1.SnapshotAPISpec
-
-	snap.Kind = "VolumeSnapshot"
-	snap.APIVersion = "v1"
-	snap.Metadata.Name = snapName
-	snap.Spec.VolumeName = volName
-
-	//Marshal serializes the value provided into a YAML document
-	yamlValue, _ := yaml.Marshal(snap)
-
-	url := GetURL() + "/latest/snapshots/create/"
-	req, err := http.NewRequest("POST", url, bytes.NewBuffer(yamlValue))
-	if err != nil {
-		return err
-	}
-
-	req.Header.Add("Content-Type", "application/yaml")
-	req.Header.Set("namespace", namespace)
-
-	c := &http.Client{
-		Timeout: http_timeout,
-	}
-	resp, err := c.Do(req)
-	if err != nil {
-		return err
-	}
-	defer resp.Body.Close()
-	body, err := ioutil.ReadAll(resp.Body)
-
-	if err != nil {
-		return err
-	}
-
-	code := resp.StatusCode
-	if err == nil && code != http.StatusOK {
-		return fmt.Errorf(string(body))
-	}
-	if code != http.StatusOK {
-		return fmt.Errorf("Server status error: %v ", http.StatusText(code))
-	}
-	return nil
+	snap := v1.VolumeSnapshot{
+		TypeMeta: v1.TypeMeta{
+			Kind:       "VolumeSnapshot",
+			APIVersion: "v1",
+		},
+		Metadata: v1.ObjectMeta{
+			Name: snapName,
+		},
+		Spec: v1.VolumeSnapshotSpec{
+			VolumeName: volName,
+		},
+	}
+
+	// Marshal serializes the values
+	jsonValue, _ := json.Marshal(snap)
+	_, err := postRequest(GetURL()+"/latest/snapshots/create/", jsonValue, namespace, true)
+	return err
 }
 
 // RevertSnapshot reverts a snapshot of volume by API request to m-apiserver
 func RevertSnapshot(volName string, snapName string, namespace string) error {
-
-	_, err := GetStatus()
-	if err != nil {
-		return err
-	}
-
-	var snap v1.SnapshotAPISpec
-
-	snap.Kind = "VolumeSnapshot"
-	snap.APIVersion = "v1"
-	snap.Metadata.Name = snapName
-	snap.Spec.VolumeName = volName
-
-	//Marshal serializes the value provided into a YAML document
-	yamlValue, _ := yaml.Marshal(snap)
-
-	url := GetURL() + "/latest/snapshots/revert/"
-	req, err := http.NewRequest("POST", url, bytes.NewBuffer(yamlValue))
-	if err != nil {
-		return err
-	}
-
-	req.Header.Add("Content-Type", "application/yaml")
-	req.Header.Set("namespace", namespace)
-	c := &http.Client{
-		Timeout: http_timeout,
-	}
-	resp, err := c.Do(req)
-	if err != nil {
-		return err
-	}
-	defer resp.Body.Close()
-
-	_, err = ioutil.ReadAll(resp.Body)
-	if err != nil {
-		return err
-	}
-	code := resp.StatusCode
-
-	if code != http.StatusOK {
-		return fmt.Errorf("Server status error: %v", http.StatusText(code))
-	}
-	return nil
```

Comment:
can you please mention why this code has been removed in PR template, i find it difficult to understand.